packages/engine-core/src/data-proxy/DataProxyEngine.ts

```diff
@@ -34,24 +38,30 @@ export class DataProxyEngine extends Engine {
     this.env = this.config.env ?? {}
     this.inlineSchema = config.inlineSchema ?? ''
     this.inlineDatasources = config.inlineDatasources ?? {}
+    this.inlineSchemaHash = config.inlineSchemaHash ?? ''
+    this.clientVersion = config.clientVersion ?? 'unknown'
 
     this.logEmitter = new EventEmitter()
     this.logEmitter.on('error', () => {})
 
-    this.initPromise = this.init()
-  }
-
-  /**
-   * !\ Asynchronous constructor that inits the properties marked with `!`.
-   * So any function that uses such a property needs to await `initPromise`.
-   */
-  private async init() {
-    // we set the network stuff up for the engine to make http calls to the proxy
     const [host, apiKey] = this.extractHostAndApiKey()
-    this.schemaHash = await createSchemaHash(this.inlineSchema)
-    this.clientVersion = getClientVersion(this.config)
+    this.remoteClientVersion = getClientVersion(this.config)
     this.headers = { Authorization: `Bearer ${apiKey}` }
     this.host = host
+
+    const promise = Promise.resolve() // hack for cloudflare
+    this.pushPromise = promise.then(() => this.pushSchema())
+  }
+
+  private async pushSchema() {
+    const res = await request(this.url('schema'), {
+      method: 'HEAD',
+      headers: this.headers,
+    })
+
+    if (res.status === 404) {
```

Comment:
This should check the body https://github.com/prisma/prisma/pull/9893/files#diff-364e4b652a8a462bff6c73f2294d7bc950f40d84eaa1515938c75a189b87772bR25